pkg/util/overridemanager/overridemanager.go

```diff
@@ -153,29 +159,41 @@ func (o *overrideManagerImpl) getMatchingClusterOverridePolicies(policies []poli
 		}
 	}
 
-	clusterMatchingPolicies := make([]policyv1alpha1.ClusterOverridePolicy, 0)
+	clusterMatchingPolicyOverriders := make([]policyOverriders, 0)
 	for _, policy := range resourceMatchingPolicies {
-		if policy.Spec.TargetCluster == nil {
-			clusterMatchingPolicies = append(clusterMatchingPolicies, policy)
-			continue
+		overrideRules := policy.Spec.OverrideRules
+		// NOTE(iawia002): This is for backward compatibility, TargetCluster and Overriders will be deprecated and
+		// we will not allow TargetCluster or Overriders and OverrideRules to co-exist (will be restricted via webhook).
```

Comment:
```suggestion
		overrideRules := policy.Spec.OverrideRules
		// Since the tuple of '.spec.TargetCluster' and '.spec.Overriders' can not co-exist with '.spec.OverrideRules'
		// (guaranteed by webhook), so we only look '.spec.OverrideRules' here.
```

Very clever way! Just propose the comments.

Reply:
Updated.